week1/citibike.sh

```diff
@@ -1,23 +1,24 @@
-#!/bin/bash
-#
-# add your solution after each of the 10 comments below
-#
-
-# count the number of unique stations
-
-# count the number of unique bikes
-
-# count the number of trips per day
-
-# find the day with the most rides
-
-# find the day with the fewest rides
-
-# find the id of the bike with the most rides
-
-# count the number of rides by gender and birth year
-
-# count the number of trips that start on cross streets that both contain numbers (e.g., "1 Ave & E 15 St", "E 39 St & 2 Ave", ...)
-
-
-# compute the average trip duration
+#!/bin/bash
+#
+# add your solution after each of the 10 comments below
+#
+
+# count the number of unique stations
+cut -d, -f4 201402-citibike-tripdata.csv | sort | uniq | wc -l
+# count the number of unique bikes
+cut -d, -f12 201402-citibike-tripdata.csv | sort | uniq | wc -l
+# count the number of trips per day
+cut -d, -f2 201402-citibike-tripdata.csv | cut -c2-11 | sort | uniq -c
+# find the day with the most rides
+cut -d, -f2 201402-citibike-tripdata.csv | cut -c2-11 | sort | uniq -c | cort -nr | head -n1 | cut -c3-7
+# find the day with the fewest rides
+cut -d, -f2 201402-citibike-tripdata.csv | cut -c2-11 | sort | grep '[0-9]' | uniq -c | sort -nr | tail -n1 | cut -c3-7

```

Comment:
The final cut here at the end causes this line to report the number of rides instead of the day. Without that last expression it's fine so we'll leave it for today.